pkg/deploy/dev-workspace/dev_workspace.go

```diff
@@ -148,37 +103,58 @@ func ReconcileDevWorkspace(deployContext *deploy.DeployContext) (bool, error) {
 		return true, nil
 	}
 
-	// check if DW exists on the cluster
-	devWorkspaceWebhookExists, err := deploy.Get(
-		deployContext,
-		client.ObjectKey{Name: DevWorkspaceWebhookName},
-		&admissionregistrationv1.MutatingWebhookConfiguration{},
-	)
+	isCreated, err := createDwNamespace(deployContext)
 	if err != nil {
 		return false, err
 	}
-
-	if devWorkspaceWebhookExists {
-		// if DW exists then check if version matches
-		if err := checkWebTerminalSubscription(deployContext); err != nil {
+	if !isCreated {
+		namespace := &corev1.Namespace{}
+		err := deployContext.ClusterAPI.NonCachedClient.Get(context.TODO(), types.NamespacedName{Name: DevWorkspaceNamespace}, namespace)
+		if err != nil {
 			return false, err
 		}
-	}
 
-	for _, syncItem := range syncItems {
-		_, err := syncItem(deployContext)
-		if !util.IsTestMode() {
+		namespaceOwnershipAnnotation := namespace.GetAnnotations()[deploy.CheEclipseOrgNamespace]
+		if namespaceOwnershipAnnotation == "" {
+			// don't manage DWO if namespace is create by someone else not but not Che Operator
+			return true, err
+		}
+
+		// if DWO is managed by another Che, check if we should take control under it after possible removal
+		if namespaceOwnershipAnnotation != deployContext.CheCluster.Namespace {
+			isOnlyOneOperatorManagesDWResources, err := isOnlyOneOperatorManagesDWResources(deployContext)
+			if err != nil {
+				return false, err
+			}
+			if !isOnlyOneOperatorManagesDWResources {
+				// Don't take a control over DWO if CheCluster in another CR is handling it
+				return true, nil
+			}
+			namespace.GetAnnotations()[deploy.CheEclipseOrgNamespace] = deployContext.CheCluster.Namespace
+			_, err = deploy.Sync(deployContext, namespace)
 			if err != nil {
 				return false, err
 			}
 		}
 	}
 
-	if !devWorkspaceWebhookExists && !util.IsTestMode() {
-		// the webhook did not exist in the cluster
-		// this means that we're installing devworkspace and therefore need to restart
-		// so that devworkspace support can initialize during the operator startup
-		afterInstall()
+	beforeSyncState := devworkspace.GetDevworkspaceState(deployContext.ClusterAPI.Scheme, org.AsV2alpha1(deployContext.CheCluster))
+
+	for _, syncItem := range syncItems {
+		_, err := syncItem(deployContext)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	afterSyncState := devworkspace.GetDevworkspaceState(deployContext.ClusterAPI.Scheme, org.AsV2alpha1(deployContext.CheCluster))
+
+	if !util.IsTestMode() {
+		if beforeSyncState != afterSyncState {
```

Comment:
@metlos @sparkoo I need your eyes here.
The problem I see with this solution - unnecessary restart may happen when several CheCluster exists, and one got DevWorkspace disabled, but it's really a corner case I believe.